File: src/main/java/de/steklopod/sec08/Lec01StartWith.java
package de.steklopod.sec08;

import de.steklopod.utils.Util;
import de.steklopod.sec08.helper.NameGenerator;

public class Lec01StartWith {

    public static void main(String[] args) {


        NameGenerator generator = new NameGenerator();
        generator.generateNames()
                .take(2)
                .subscribe(Util.subscriber("sam"));

        generator.generateNames()
                .take(2)
                .subscribe(Util.subscriber("mike"));

        generator.generateNames()
                .take(3)
                .subscribe(Util.subscriber("Jake"));

        generator.generateNames()
                .filter(n -> n.startsWith("A"))
                .take(2)
                .subscribe(Util.subscriber("Marshal"));


    }


}

File: src/main/kotlin/de/steklopod/sec08/Lec01StartWith.kt
package de.steklopod.sec08

import de.steklopod.sec08.helper.NameGenerator
import de.steklopod.utils.Util.subscriber

object Lec01StartWith {
    @JvmStatic
    fun main(args: Array<String>) {
        val generator = NameGenerator()

        generator.generateNames()
            .take(2)
            .subscribe(subscriber("sam"))

        generator.generateNames()
            .take(2)
            .subscribe(subscriber("mike"))

        generator.generateNames()
            .take(3)
            .subscribe(subscriber("Jake"))

        generator.generateNames()
            .filter { n: String -> n.startsWith("A") }
            .take(2)
            .subscribe(subscriber("Marshal"))
    }
}
